lfs/filters.go

```diff
@@ -0,0 +1,76 @@
+package lfs
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/github/git-lfs/git"
+)
+
+var (
+	valueRegexp = regexp.MustCompile("\\Agit[\\-\\s]media")
+)
+
+// A Filter represents a git-configurable attribute of the type "filter". It
+// holds a name and a value.
+type Filter struct {
+	Name  string
+	Value string
+}
+
+// Install installs the filter in context. It returns any errors it encounters
+// along the way. If the value for this filter is set, but not equal to the
+// value that is given, then an exception will be returned.  If `force` is
+// passed as true in that same situtation, the value will be overridden.  All
+// other cases will pass.
+func (f *Filter) Install(force bool) error {
+	key := fmt.Sprintf("filter.lfs.%s", f.Name)
+
+	currentValue := git.Config.Find(key)
+	if force || f.shouldReset(currentValue) {
+		git.Config.UnsetGlobal(key)
+		git.Config.SetGlobal(key, f.Value)
+
+		return nil
+	} else if currentValue != f.Value {
+		return fmt.Errorf("The %s filter should be \"%s\" but is \"%s\"",
+			f.Name, f.Value, currentValue)
+	}
+
+	return nil
+}
+
+// shouldReset determines whether or not a value is resettable given its current
+// value on the system. If the value is empty (length = 0), then it will pass.
+// Otherwise, it will pass if the below regex matches.
+func (f *Filter) shouldReset(value string) bool {
+	if len(value) == 0 {
+		return true
+	}
+
+	return valueRegexp.MatchString(value)
+}
+
+// The Filters type represents a set of filters to install on the system. It
+// exists purely for the convenience of being able to stick the Teardown method
+// somewhere where it makes sense.
+type Filters []*Filter
```

Comment:
> It exists purely for the convenience of being able to stick the Teardown method somewhere where it makes sense.

I think it makes sense to just have functions like `SetupFilters([]*Filter)` and `TeardownFilters([]*Filter)`, too. I'm not sure how much is gained by declaring a `Filters` type.